src/main/java/seedu/address/logic/commands/ingredientcommands/IngredientViewSingleCommand.java

```diff
@@ -0,0 +1,113 @@
+package seedu.address.logic.commands.ingredientcommands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INGREDIENT;
+
+import java.util.Optional;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.SetCommand;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ingredient.Ingredient;
+import seedu.address.model.ingredient.IngredientName;
+
+/**
+ * View the level of a single specific ingredient.
+ */
+public class IngredientViewSingleCommand extends Command {
+
+    public static final String COMMAND_WORD = "i-view-single";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " :view a single ingredient level in tCheck."
+            + "Parameters: "
+            + PREFIX_INGREDIENT
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_INGREDIENT + "Milk ";
+
+    public static final String MESSAGE_SUCCESS = "Here is the ingredient and its level: %1$s";
+    private final IngredientName target;
+    private final ViewIngredientDescriptor viewIngredientDescriptor;
+
+    /**
+     * Constructs an IngredientViewSingle command with the given ingredient name.
+     */
+    public IngredientViewSingleCommand(IngredientName target, ViewIngredientDescriptor viewIngredientDescriptor) {
+        requireNonNull(target);
+        requireNonNull(viewIngredientDescriptor);
```

Comment:
Can be combined by using `requireAllNonNull`, however, this is also fine.

Reply:
Noted with thanks